### PR TITLE
docs: Additional example for custom job template.

### PR DESCRIPTION
**Pull request by abates**
# What's Changed
Added an example job with a custom template that provides an overridden form element.

In this case, an example job is included that overrides a `StringVar` field (which would typically use a standard `<input type="text">` field) to display a `<textarea>` field in the custom template:

Screenshot 2024-07-31 at 12 29 33 PM

# TODO
- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] ~Documentation Updates (when adding/changing features)~
- [x] Example App Updates (when adding/changing features)
- [ ] ~Outline Remaining Work, Constraints from Design~

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Looks solid as an example. Would you be willing to add a test case to `nautobot/extras/tests/test_views.py` that loads the job-run view for this new job and uses `self.assertTemplateUsed("example_app/custom_job_form.html")` to assert that the custom template is in fact rendered?

**Comment by abates** [comment]
I will add the test, I also want to update the code comments slightly for clarity (based on feedback from @itdependsnetworks). I'll try to get this done today or tomorrow and mark the PR as ready.

File: examples/example_app/example_app/templates/example_app/custom_job_form.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    Note that we are not calling `block.super` here because we don't want any
+    part of the parent job_form block to be rendered (in this case).
+-->
+<div class="form-group">
+    {% with field=job_form.fields.custom_job_data %}
+        {% with field_value=job_form.initial.custom_job_data|default:field.initial %}
+    <label class="col-md-3 control-label required" for="id_custom_job_data">{{ field.label }}</label>
+    <div class="col-md-9">
+        <textarea id="id_custom_job_data" name="custom_job_data">{{ field_value }}</textarea>
```
Comment (glennmatthews) [inline examples/example_app/example_app/templates/example_app/custom_job_form.html:19]: Might be that adding `class="form-control"` to the textarea might improve its rendering - in the provided screenshot it doesn't look like it's styled consistently with the rest of the form at present.

Comment (abates) [reply examples/example_app/example_app/templates/example_app/custom_job_form.html]: Done.

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**